src/protagonist/Portal/Features/Batches/Requests/ParseCsv.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Security.Claims;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using API.Client;
+using DLCS.HydraModel;
+using Hydra.Collections;
+using MediatR;
+using Microsoft.AspNetCore.Http;
+using CsvHelper;
+using CsvHelper.Configuration.Attributes;
+using CsvHelper.TypeConversion;
+using DLCS.Web.Auth;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Portal.Settings;
+using MissingFieldException = CsvHelper.MissingFieldException;
+
+namespace Portal.Features.Batches.Requests;
+
+public class ParseCsv : IRequest<ParseCsvResult>
+{
+    public IFormFile File { get; set; }
+}
+
+public class ParseCsvResult
+{
+    [JsonIgnore] public bool IsSuccess { get; set; }
+
+    public string[]? Errors { get; set; }
+
+    public static ParseCsvResult Failure(string[] errors)
+    {
+        return new ParseCsvResult { IsSuccess = false, Errors = errors };
+    }
+    
+    public static ParseCsvResult Failure(string error)
+    {
+        return new ParseCsvResult { IsSuccess = false, Errors = new []{ error } };
+    }
+    
+    public static ParseCsvResult Success()
+    {
+        return new ParseCsvResult { IsSuccess = true };
+    }
+}
+
+public class ParseCsvHandler : IRequestHandler<ParseCsv, ParseCsvResult>
+{
+    private readonly IDlcsClient dlcsClient;
+    private readonly ILogger<ParseCsvHandler> logger;
+    private readonly int customerId;
+    private readonly int maxBatchSize;
+    
+    public ParseCsvHandler(IDlcsClient dlcsClient, IOptions<PortalSettings> portalSettings, ILogger<ParseCsvHandler> logger, ClaimsPrincipal currentUser)
+    {
+        this.dlcsClient = dlcsClient;
+        this.logger = logger;
+        customerId = (currentUser.GetCustomerId() ?? -1);
+        maxBatchSize = portalSettings.Value.MaxBatchSize;
+    }
+
+    public async Task<ParseCsvResult> Handle(ParseCsv request, CancellationToken cancellationToken)
```

Comment:
There's a lot happening in this method. It might be easier to read if it were split down into more meaningful methods - makes it easier to see what it does a glance (ParsesCsv then UploadsToDlcs) and if you need to see details you can look into methods.

```cs
public async Task<ParseCsvResult> Handle(ParseCsv request, CancellationToken cancellationToken)
{
  var (distinctRows, readErrors) = ParseCsv(request.File);

  if (readErrors.Any()) 
  {
    return ParseCsvResult.Failure(readErrors.ToArray());
  }

  var parseCsvResult = await UploadImagesToDlcs(distinctRows);
  return parseCsvResult;
}
```

A note on the last `var parseCsvResult` - returning that from a method that doesn't parse highlights that name is a little out.